homeassistant/components/arcam_fmj/config_flow.py

```diff
@@ -16,12 +67,75 @@ class ArcamFmjFlowHandler(config_entries.ConfigFlow):
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
 
-    async def async_step_import(self, import_config):
-        """Import a config entry from configuration.yaml."""
-        entries = self.hass.config_entries.async_entries(DOMAIN)
-        import_key = _GETKEY(import_config)
-        for entry in entries:
-            if _GETKEY(entry.data) == import_key:
-                return self.async_abort(reason="already_setup")
+    async def _async_set_unique_id_and_update(self, host, port, uuid):
+        config = {
+            CONF_HOST: host,
+            CONF_PORT: port,
+        }
+        entry = await self.async_set_unique_id(uuid)
+        if not entry:
+            return
+
+        if config == entry.data:
+            raise AbortFlow("already_setup")
+        self.hass.config_entries.async_update_entry(entry, data=config)
+        raise AbortFlow("updated_instance")
```

Comment:
This works exactly the same:

```suggestion
        await self._abort_if_unique_id_configured(config)
```